### PR TITLE
Retro UI updates, pre-Aug (ready)

### PR DESCRIPTION
**Pull request by ackernaut**
Addresses “design fixes” from #2022 

## design fixes
- [x] Add a reflection button should have userSelect: none
- [x] the scrollbar on the right is always present (probably a linux-only thing) (already done in previous PR)
- [x] yank number of cards per group (adds pencil to indicate it can be edited)
- [x] maybe remove column headers on vote stage, maybe move to a LTR grid
- [x] agenda item overflow (already done in previous PR)
- [x] too many scroll bars in discuss phase (already done in previous PR)
- [x] put add a reflection at the top instead of the bottom (less than ideal but it’s a start — here’s a [Figma prototype](https://www.figma.com/proto/TLD1OTiMLONajdyYOaErrG/Retro-v1.1.0?node-id=1568%3A9769&scaling=scale-down) to illustrate a more ideal UX)
- [x] make everyone else's reflections slightly darker during the reflect phase
- [x] consider yanking flag and showing users with no tasks in retro summary (because of attendance)
- [x] clicking end meeting is too easy

and

- [x] users can select text from read-only reflection cards (the UX story = perhaps folks want to copy and quote the text in their take-away tasks)
- [x] quick fix for #2035 (improved dash and meeting layout to be addressed with more effort)

### Test

- [ ] User cannot select text on Add Reflection or Add Task button
- [ ] Scroll bars are now set to `auto` thanks to a previous PR that adds `<ScrollableBlock />`
- [ ] User does not see a count by group theme labels
- [ ] User sees a light pencil icon to indicate that theme can be edited, only during the Group phase when it has not been completed
- [ ] Sidebar nav was fixed thanks to previous PR
- [ ] No more column headers during VOTE (temporary solution as we want to explore a grid layout, to be done later)
- [ ] Add reflection control is at the top (not ideal because the cards get added to the bottom, but having the control at the bottom was not ideal either — temporary as we are planning to add the card inline then drop it into the top of the list, not the bottom — here’s [the better concept](https://www.figma.com/proto/TLD1OTiMLONajdyYOaErrG/Retro-v1.1.0?node-id=1568%3A9769&scaling=scale-down))
- [ ] Anonymous cards do not have a box-shadow and are slightly transparent so my reflections stand out more during the REFLECT phase
- [ ] Summary shows users with no tasks in order to also show more attendance info for team
- [ ] End meeting button is far right, next topic button is centered, until the last topic where the end meeting button is centered
- [ ] when the viewport is less than 1280px wide the user can scroll LTR and the layout is in tact and usable (the quick fix)

### CODE REVIEW

**Comment by ackernaut** [comment]
any volunteers to review this one @mattkrick @jordanh ?

**Comment by ackernaut** [comment]
This is passing build but not codecov FYI.

File: src/universal/components/ReflectionEditorWrapper.js [hunk 1]
```diff
@@ -176,7 +176,12 @@ class ReflectionEditorWrapper extends Component<Props> {
 
   render() {
     const {ariaLabel, editorState, anonEditing, isBlurred, onBlur, onFocus, placeholder, readOnly} = this.props;
-    const userSelect = readOnly ? 'none' : 'text';
+    // Folks may want to copy text from reflection cards to quote in task cards, so going to allow unless blurred (TA)
+    const userSelect = (isBlurred || anonEditing) ? 'none' : 'text';
```
Comment (mattkrick) [inline src/universal/components/ReflectionEditorWrapper.js]: we need to fix or remove the anonEditing variable. it's defined as !isBlurred, so it doesn't provide any real value.

File: src/universal/components/ReflectionGroupHeader.js [hunk 2]
```diff
@@ -38,18 +39,36 @@ const ReflectionCount = styled('span')({
...
+});
+
 const Spacer = styled('div')({width: ui.votingCheckmarksWidth});
 
 const ReflectionGroupHeader = (props: Props) => {
   const {meeting, reflectionGroup} = props;
   const {reflections} = reflectionGroup;
-  const {localPhase: {phaseType}} = meeting;
+  const {localStage, localPhase: {phaseType}} = meeting;
+  console.log('meeting');
```
Comment (mattkrick) [inline src/universal/components/ReflectionGroupHeader.js]: nuke

File: src/universal/components/ReflectionGroupHeader.js [hunk 3]
```diff
@@ -38,18 +39,36 @@ const ReflectionCount = styled('span')({
...
+  console.dir(meeting);
+  const groupPhaseNotComplete = phaseType === GROUP && localStage.isComplete === false;
+  const showCount = false; // local flag, testing new UI (TA)
```
Comment (mattkrick) [inline src/universal/components/ReflectionGroupHeader.js]: nuke, let's just get rid of the count

File: src/universal/components/ReflectionGroupHeader.js [hunk 4]
```diff
@@ -38,18 +39,36 @@ const ReflectionCount = styled('span')({
...
 const Spacer = styled('div')({width: ui.votingCheckmarksWidth});
 
 const ReflectionGroupHeader = (props: Props) => {
   const {meeting, reflectionGroup} = props;
   const {reflections} = reflectionGroup;
-  const {localPhase: {phaseType}} = meeting;
+  const {localStage, localPhase: {phaseType}} = meeting;
+  console.log('meeting');
+  console.dir(meeting);
+  const groupPhaseNotComplete = phaseType === GROUP && localStage.isComplete === false;
```
Comment (mattkrick) [inline src/universal/components/ReflectionGroupHeader.js]: kinda misleading variable name. if i'm on the vote stage, then this is false.
consider `canEdit` or something

File: src/universal/components/RetroDiscussPhase.js [hunk 5]
```diff
@@ -157,29 +161,39 @@ const RetroDiscussPhase = (props: Props) => {
...
-          buttonStyle="flat"
-          colorPalette="dark"
-          icon="arrow-circle-right"
-          iconLarge
-          iconPalette="warm"
-          iconPlacement="right"
-          label={'End Meeting'}
-          onClick={endMeeting}
-        />
+      <MeetingControlBar style={{justifyContent: 'space-between'}}>
```
Comment (mattkrick) [inline src/universal/components/RetroDiscussPhase.js]: consider `SpacedMeetingControlBar = styled(MeetingControlBar)({...})`

File: src/universal/components/ReflectionEditorWrapper.js (hunk 1 above)
Comment (ackernaut) [reply src/universal/components/ReflectionEditorWrapper.js]: Yeah, the only catch that I’m not sure to work around is for a styling detail. I want normal editing styles, blurred styles, and hint color style for “Somebody is typing…” — basically I need to know if the styles are for the anon card or not and if that card is blurred or is “Somebody is typing…” hint. I don’t want any user select for the anon card. I will try to simplify it a little.

**Comment by ackernaut** [comment]
### Test No. 1 (in progress)

- [ ] User cannot select text on Add Reflection or Add Task button
- [ ] Scroll bars are now set to `auto` thanks to a previous PR that adds `<ScrollableBlock />`
- [ ] User does not see a count by group theme labels
- [ ] User sees a light pencil icon to indicate that theme can be edited, only during the Group phase when it has not been completed
- [ ] Sidebar nav was fixed thanks to previous PR
- [ ] No more column headers during VOTE (temporary solution as we want to explore a grid layout, to be done later)
- [ ] Add reflection control is at the top (not ideal because the cards get added to the bottom, but having the control at the bottom was not ideal either — temporary as we are planning to add the card inline then drop it into the top of the list, not the bottom — here’s [the better concept](https://www.figma.com/proto/TLD1OTiMLONajdyYOaErrG/Retro-v1.1.0?node-id=1568%3A9769&scaling=scale-down))
- [ ] Anonymous cards do not have a box-shadow and are slightly transparent so my reflections stand out more during the REFLECT phase
- [ ] Summary shows users with no tasks in order to also show more attendance info for team
- [ ] End meeting button is far right, next topic button is centered, until the last topic where the end meeting button is centered
- [ ] when the viewport is less than 1280px wide the user can scroll LTR and the layout is in tact and usable (the quick fix)

**Comment by ackernaut** [comment]
@mattkrick I think this is ready to be merged, it just doesn’t pass codecov

File: src/universal/components/ReflectionEditorWrapper.js (hunk 1 above)
Comment (mattkrick) [reply src/universal/components/ReflectionEditorWrapper.js]: gotcha, in that case in the AnonymouseReflectionCard i'd pass down an `isAnonymous` prop. then, you can just say userSelect = isAnonymous ? 'none' : 'text';

**Pull request merged**